src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -66,45 +69,24 @@ grpc_core::RefCountedPtr<grpc_auth_context> local_auth_context_create(
   return ctx;
 }
 
+static void uri_to_sockaddr(const char* uri_str, grpc_resolved_address* addr) {
+  absl::StatusOr<grpc_core::URI> uri = grpc_core::URI::Parse(uri_str);
+  if (!uri.ok()) {
+    gpr_log(GPR_ERROR, "%s", uri.status().ToString().c_str());
+    GPR_ASSERT(uri.ok());
+  }
+  if (!grpc_parse_uri(*uri, addr)) memset(addr, 0, sizeof(*addr));
+}
+
 void local_check_peer(tsi_peer peer, grpc_endpoint* ep,
                       grpc_core::RefCountedPtr<grpc_auth_context>* auth_context,
                       grpc_closure* on_peer_checked,
                       grpc_local_connect_type type) {
-  int fd = grpc_endpoint_get_fd(ep);
-  grpc_resolved_address resolved_addr;
-  memset(&resolved_addr, 0, sizeof(resolved_addr));
-  resolved_addr.len = GRPC_MAX_SOCKADDR_SIZE;
-  bool is_endpoint_local = false;
-  if (getsockname(fd, reinterpret_cast<grpc_sockaddr*>(resolved_addr.addr),
-                  &resolved_addr.len) == 0) {
-    grpc_resolved_address addr_normalized;
-    grpc_resolved_address* addr =
-        grpc_sockaddr_is_v4mapped(&resolved_addr, &addr_normalized)
-            ? &addr_normalized
-            : &resolved_addr;
-    grpc_sockaddr* sock_addr = reinterpret_cast<grpc_sockaddr*>(&addr->addr);
-    // UDS
-    if (type == UDS && grpc_is_unix_socket(addr)) {
```

Comment:
This check is missing from the new code.

Reply:
Good catch. I thought this check will be enforced in other places, but this turned out to be the only place checking `grpc_local_connect_type`.